pkg/restapi/oidc/operations.go

```diff
@@ -168,16 +167,26 @@ func (o *Operation) oidcCallbackHandler(w http.ResponseWriter, r *http.Request)
 		return
 	}
 
-	err = newPersistedData(o.store.users).put(user.ID, user)
-	if err != nil {
+	_, err = newPersistedData(o.store.users).getEndUser(user.Sub)
+	if err != nil && !errors.Is(err, storage.ErrValueNotFound) {
 		common.WriteErrorResponsef(w, logger,
-			http.StatusInternalServerError, "failed to persist user data: %s", err.Error())
+			http.StatusInternalServerError, "failed to query user data: %s", err.Error())
 
 		return
 	}
 
-	err = newPersistedData(o.store.tokens).put(user.ID, &endUserTokens{
-		ID:      user.ID,
+	if errors.Is(err, storage.ErrValueNotFound) {
+		err = newPersistedData(o.store.users).put(user.Sub, user)
+		if err != nil {
+			common.WriteErrorResponsef(w, logger,
+				http.StatusInternalServerError, "failed to persist user data: %s", err.Error())
+
+			return
+		}
+	}
+
+	err = newPersistedData(o.store.tokens).put(user.Sub, &endUserTokens{
```

Comment:
returned `err` of put() call is not checked here, also why not use the same put() call for `err == nil` and `errors.Is(err, storage.ErrValueNotFound)` above?

Reply:
@Baha-sk 

> returned `err` of put() call is not checked here

`err` is checked right below this line.

> also why not use the same put() call for `err == nil` and `errors.Is(err, storage.ErrValueNotFound)` above?

This statement is saving `endUserTokens`. The one above is saving `endUser`.

Reply:
I see, it was in the unchanged code section that I haven't opened. Sounds good then.